features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/members/RoomMemberListPresenter.kt

```diff
@@ -57,6 +62,13 @@ class RoomMemberListPresenter @Inject constructor(
             value = room.canInvite().getOrElse { false }
         }
 
+        val canDisplayBannedUsers by produceState(initialValue = false) {
+            room.membersStateFlow
+                .map { members -> members.roomMembers()?.any { it.membership == RoomMembershipState.BAN }.orFalse() }
+                .onEach { hasBannedUsers -> value = hasBannedUsers && room.canBan().getOrElse { false } && !room.isDm }
```

Comment:
The test `!room.isDm` is a bit late, maybe check it before checking the roomMembers?
Note that on a DM there should not be a lot of user.

(I am wondering why the fact that a DM should be taken into account to display or not the banned users, it seems that it may lead to problem if the room actually has banned users. Similarly, I do not see the point to hide the banned users to user who cannot unban. Sorry to rise the point during the PR review, they have maybe been discussed with product/design. It it is the case, just ignore me :) ).

Reply:
> The test !room.isDm is a bit late, maybe check it before checking the roomMembers?

True, it would be a lot faster that way.

> (I am wondering why the fact that a DM should be taken into account to display or not the banned users

A DM in this context is a direct room with 2 users, you and the other user. So if you are banned you won't be able to access this view and if the other user is banned you won't be able to talk to anyone anyway, so it would be easier to just leave the room.